mdal/frmts/mdal_ply.cpp

```diff
@@ -0,0 +1,445 @@
+  //
+  do
+  {
+    //
+    // all element blocks block in the header and enumerate the number of members and the properties
+    //
+    if ( startsWith( line, "element" ) )
+    {
+      chunks = split( line, ' ' );
+      if ( ( chunks.size() != 3 ) )
+      {
+        MDAL::Log::error( MDAL_Status::Err_IncompatibleMesh, name(), meshFile + " invalid number of vertexes" );
+        return nullptr;
+      }
+      MDAL::DriverPly::element element;
+      element.name = chunks[1];
+      element.size = MDAL::toSizeT( chunks[2] );
+      do
+      {
+        if ( !std::getline( in, line ) )
+        {
+          MDAL::Log::error( MDAL_Status::Err_IncompatibleMesh, name(), meshFile + " Header is corrupt" );
+          return nullptr;
+        }
+        if ( startsWith( line, "property" ) )
+        {
+          chunks = split( line, ' ' );
+          switch ( chunks.size() )
+          {
+            case 3:
+              element.properties.push_back( chunks[2] );
+              element.types.push_back( chunks[1] );
+              element.list.push_back( false );
+              break;
+            case 5:
+              element.properties.push_back( chunks[4] );
+              element.types.push_back( chunks[3] );
+              element.list.push_back( true );
+              break;
+            default:
+              MDAL::Log::error( MDAL_Status::Err_IncompatibleMesh, name(), meshFile + " invalid element line" + line );
+              return nullptr;
+          }
+        }
+        else
+        {
+          break;
+        }
+      }
+      while ( true );
+      elements.push_back( element );
+    }
+    //
+    // if end - stop looping
+    else if ( startsWith( line, "end_header" ) )
+    {
+      break;
+    }
+    //
+    //if binary - give up
+    //
+    else if ( startsWith( line, "binary" ) )
+    {
+      MDAL::Log::error( MDAL_Status::Err_IncompatibleMesh, name(), meshFile + " only ASCII format PLY files are supported" );
+      return nullptr;
+    }
+
+    //
+    // if "comment crs" assume that the rest is the crs data
+    //
+    else if ( startsWith( line, "comment crs " ) )
+    {
+      line.erase( 0, 12 );
+      proj = line;
+      if ( !std::getline( in, line ) )
+      {
+        MDAL::Log::error( MDAL_Status::Err_IncompatibleMesh, name(), meshFile + " Header is corrupt" );
+        return nullptr;
+      }
+    }
+
+    //
+    // probably a comment line
+    //
+    else
+    {
+      if ( !std::getline( in, line ) )
+      {
+        MDAL::Log::error( MDAL_Status::Err_IncompatibleMesh, name(), meshFile + " Header is corrupt" );
+        return nullptr;
+      }
+    }
+  }
+  while ( true );
+
+  Vertices vertices( 0 );
+  Faces faces( 0 );
+  Edges edges( 0 );
+  size_t maxSizeFace = 0;
+  size_t faceSize = 0;
+
+  //datastructures that will contain all of the datasets, categorised by vertex, face and edge datasets
+  std::vector<std::vector<double>*> vertexDatasets; // conatins the data
```

Comment:
why do you use pointer of vector<double>? Why not simply used `std::vector<std::vector<double>>?

Reply:
Mostly because with all of the later updates coming to both dimensions of the array it felt more efficient to just hold the pointer.

But to be honest, I don't do a lot of cpp, So if you felt it to be better I could change it - it could make things a little simpler later on.